cmd/autoheal/healer.go

```diff
@@ -200,6 +208,52 @@ func (h *Healer) Run(stopCh <-chan struct{}) error {
 	return nil
 }
 
+// onChangeEvent reload config files and
+// refresh rules in rules cache (by sending "Deleted" + "Added" to queue).
+//
+func (h *Healer) onChangeEvent(e *config.ChangeEvent) {
+	mutex := &sync.Mutex{}
```

Comment:
If you create the mutex inside the function then it has no effect, as each goroutine will use its own mutex. If you really need to make sure that two events aren't processed in parallel then you will need to do something like this:

```go
type Healer {
  ...
  // Mutex to prevent simultaneous reloads of configuration.
  reloadMutex *sync.Mutex
  ...
}

func (h *Healer) onChangeEvent(e *config.ChangeEvent) {
  h.reloadMutex.Lock()
  defer h.reloadMutex.Unlock()

  ...
}
```